core/shared/src/main/scala/cats/effect/unsafe/Scheduler.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats.effect.unsafe
+
+import scala.concurrent.duration.FiniteDuration
+
+import java.util.concurrent.ScheduledExecutorService
+
+trait Scheduler {
+
+  /**
+   * Schedules a side-effect to run after the delay interval. Produces
+   * another side-effect which cancels the scheduling.
+   */
+  def sleep(delay: FiniteDuration, task: Runnable): Runnable
+
+  def nowMillis(): Long
+
+  def monotonicNanos(): Long
+}
+
+object Scheduler {
+  def fromScheduledExecutor(scheduler: ScheduledExecutorService): Scheduler =
```

Comment:
I was thinking we should put this into a `SchedulerPlatform` companion for the JVM, and have a different one for JS, since `ScheduledExecutorService` does not link on ScalaJS.

Reply:
Cool, those changes are both in.